docs/src/guide/invariants/index.md

```diff
@@ -0,0 +1,330 @@
+Verification failed:
+  when ALL participants are honest
+  of theorem: assert
+  msg: "balance sufficient for transfer"
+  at ./index.rsh:94:31:application
+
+  // Violation Witness
+
+  const UInt.max = 4;
+
+  const wager/297 = "Alice".interact.wager;
+  //    ^ could = 1
+  //      from: ./index.rsh:30:14:property binding
+  const netBalance/330 = <loop variable>;
+  //    ^ could = 0
+  //      from: ./index.rsh:61:5:while
+
+  // Theorem Formalization
+
+  const v452 = (2 * wager/297) <= netBalance/330;
+  //    ^ would be false
+  assert(v452);
+  
+  Verification failed:
+  when ALL participants are honest
+  of theorem: assert
+  msg: "balance zero at application exit"
+  at ./index.rsh:27:30:compileDApp
+
+  // Violation Witness
+
+  const UInt.max = 6;
+
+  const wager/297 = "Alice".interact.wager;
+  //    ^ could = 0
+  //      from: ./index.rsh:30:14:property binding
+  const netBalance/330 = <loop variable>;
+  //    ^ could = 1
+  //      from: ./index.rsh:61:5:while
+
+  // Theorem Formalization
+
+  const v456 = 0 == (netBalance/330 - (2 * wager/297));
+  //    ^ would be false
+  assert(v456);
+
+  Verifying when NO participants are honest
+Checked 77 theorems; 4 failures (and 2 omitted repeats) :'(
+
+For a guide to understanding verification failures, see: https://docs.reach.sh/rsh/errors/#how-to-read-verification-failures
+```
+
+The error message indicates that the balance is not sufficient for a transfer on line 94. 
+This line is far from our `{!rsh} while` loop and invariant. 
+
+Line 94 is where we transfer the winnings to Alice or Bob based on who won the round.
+
+``` rsh
+load: /examples/rps-7-loops/index.rsh
+md5: ee287e712cdfe8d91bbb038c383d25d3
+range: 93-95
+```
+
+If you don't understand the flow of the balance within your application then this error may be confusing. 
+Even though the trouble begins at the loop invariant in line 60, the error is triggered at line 94 because this is the first time we try to move the balance out of the contract. 
+
+Reading the Violation Witness may be more helpful in this situation. 
+(If you'd like to gain a better understanding of Violation Witnesses and Theorem Formalizations then read our guide, @{seclink("how-to-read-verification-failures")}).
+The verification engine is showing two possible places where the insufficient balance could have first occurred. 
+
+The first place could be from the `wager` property binding in the Participant Interact Interface. 
+The second indicates that the insufficient balance could be originating in the while loop in line 61.
+The next thing we see is the Theorem Formalization where the verification engine is showing the math of how the program fails.
+
+If you didn't already know that the balance was failing in the while loop then you'd want to investigate the lines that the Violation Witness points to. 
+
+We see a second verification failure that indicates that the balance is not zero when the application exits. 
```

Comment:
At this point, I would sample the other error and then start reference its line numbers